url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/GUI/MainFrame.java

```diff
@@ -0,0 +1,588 @@
+
+						 // sérialization de l'objet
+						oos.writeObject(dataStore);
+				    	JOptionPane.showMessageDialog(null, "La sauvegarde a été réalisée avec succès.", "Sauvegarde réussie", JOptionPane.INFORMATION_MESSAGE);
+
+					} catch (IOException e1) {
+						// TODO Auto-generated catch block
+						e1.printStackTrace();
+					}
+				}
+			});
+			btnSave.setText("Enregistrer");
+			
+			btnOpen = new JButton(new ImageIcon(ImageIO.read(new File("img/icons/folder.png"))));
+			btnOpen.setText("Ouvrir");
+			btnOpen.addActionListener(new ActionListener() {
+				
+				@Override
+				public void actionPerformed(ActionEvent e) {
+int n = JOptionPane.showConfirmDialog(null, "Voulez sauvegarder la session en cours ?", "Attention", JOptionPane.YES_NO_CANCEL_OPTION);
+					
+					switch(n)	{
+					case JOptionPane.YES_OPTION :
+						BufferedReader br;
+						try {
+							br = new BufferedReader(new FileReader(System.getProperty("user.dir") + "/dataStoreLocation.loc"));
+							String fileName = "";
+							String sCurrentLine = "";
+							while ((sCurrentLine = br.readLine()) != null) {
+								fileName = sCurrentLine;
+							}
+							File fichier =  new File(fileName) ;
+							 // ouverture d'un flux sur un fichier
+							ObjectOutputStream oos =  new ObjectOutputStream(new FileOutputStream(fichier)) ;
+
+							 // sérialization de l'objet
+							oos.writeObject(dataStore);
+					    	JOptionPane.showMessageDialog(null, "La sauvegarde a été réalisée avec succès.", "Sauvegarde réussie", JOptionPane.INFORMATION_MESSAGE);
+
+						} catch (IOException e1) {
+							// TODO Auto-generated catch block
+							e1.printStackTrace();
+						}
+						break;
+						
+						
+					case JOptionPane.NO_OPTION :
+						break;
+						
+					case JOptionPane.CANCEL_OPTION :
+						return;
+					}
+					
+					
+					JFileChooser fc = new JFileChooser();
+					fc.setFileFilter(new FileNameExtensionFilter("Emploi du Temps", "EdT"));
+					int res = fc.showOpenDialog(null);
+					File f = null;
+					if (res == JFileChooser.APPROVE_OPTION)	{
+						// On a le fichier.
+						f = fc.getSelectedFile();
+						// On sauvegarde sa localisation pour la prochaine ouverture du programme.
+						File dataStoreLocation = new File(System.getProperty("user.dir") + "/dataStoreLocation.loc");
+						BufferedWriter writer;
+						try {
+							writer = new BufferedWriter(new FileWriter(dataStoreLocation));
+							writer.write (f.getAbsolutePath()); // On écrit le chemin du DataStore
+							writer.close();
+						} catch (IOException e1) {
+							// TODO Auto-generated catch block
+							JOptionPane.showMessageDialog(null, "Impossible de sauvegarde la localisation du fichier de sauvegarde.", "Erreur de sauvegarde", JOptionPane.ERROR_MESSAGE);
+						}
+
+						// On essaie maintenant de charger le DataStore
+						try
+						{
+							FileInputStream inputFileStream = new FileInputStream(f.getAbsolutePath());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *inputFileStream*. The resource is referenced at lines 183, 184. The resource is closed at lines 185, 186. There are other execution paths that do not contain closure statements, for example, when ObjectInputStream.readObject throws an exception. Ensure *inputFileStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).